pkg/sync/kubernetes/kubernetes_sync_test.go

```diff
@@ -0,0 +1,362 @@
+package kubernetes
```

Comment:
Thanks for adding the tests. I kept meaning to add an issue to implement these and forgot to do it.

Reply:
Thanks. Yes, I added tests for easily testable parts. However, we still need coverage on K8s-backed logic. These need either mocking or true e2e tests.

Reply:
We already have the "true" e2e tests in OFO. I think that's enough.

I think though we should improve coverage in this repo as well..